hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AConfiguration.java

```diff
@@ -361,6 +362,14 @@ public void shouldBeAbleToSwitchOnS3PathStyleAccessViaConfigProperty()
       // isn't in the same region as the s3 client default. See
       // http://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html
       assertEquals(HttpStatus.SC_MOVED_PERMANENTLY, e.getStatusCode());
+    } catch (final IllegalArgumentException e) {
+      // Path style addressing does not work with AP ARNs
+      if (conf.get(ACCESS_POINT_ARN, "").isEmpty()) {
+        LOG.error("Caught unexpected exception: ", e);
+        throw e;
+      }
+
+      assertTrue(e.getMessage().contains("ARN of type accesspoint cannot be passed as a bucket"));
```

Comment:
there's a GenericTestUtils check which will rethrow the exception if it doesn't contain the string; ensures the root exception isn't lost